### PR TITLE
feat(#941): D39-2.4a — conflict resolution + branch graph SKELETON

### PR DESCRIPTION
**Pull request by zjzcpj**
Closes #941

## Summary

ADR-039 §3.5a / §3.5b / §6 Phase 3 SKELETON. Scaffolds the conflict-resolution UI (Monaco decorations + inline action buttons + merge orchestrator) and the clean-room branch-graph component. All non-pure-helper bodies throw `Error("TODO: D39-2.4b — …")`. Pure helpers (`parseConflictRegions`, `maxLane`, `buildShaIndex`, `colorForIndex`, `PALETTE`) are fully implemented and tested so D39-2.4b has them as load-bearing primitives.

### Conflict resolution (ADR §3.5a)

- `MergeFlow.tsx` — state-machine doc (IDLE → IN_FLIGHT → {FAST-FORWARD, CLEAN, CONFLICT, ERROR} → COMPLETE/ABORT), literal copy strings, `gitSlice.mergeInProgress` lifecycle, edge cases (network failure mid-merge, user closes modal during conflict, current-branch-changes mid-flow). Renders a stub div when `isOpen`.
- `ConflictResolveView.tsx` — per-file Unresolved/Resolved badge contract, Mark Resolved / Complete Merge / Abort Merge wiring plan, `aria-disabled` rule for "Complete" until all files resolved, "no-markers detected → auto-prompt Mark Resolved" UX edge case.
- `ConflictMarkerDecoration.ts` — Monaco decoration provider scaffold. `parseConflictRegions(content)` is fully IMPLEMENTED (no Monaco dependency): handles 2-way + diff3 styles, falls back to default labels, strips CRLF, recovers from nested `<<<<<<<` defensively, discards unclosed-at-EOF partials. 9 vitest assertions cover all of these.

### Branch graph (ADR §3.5b clean-room)

- `laneAssign.ts` — FULL pseudocode transcribed verbatim from ADR §3.5b sketch. 11-edge-case enumeration (empty, single, root-in-middle, linear, two-way merge, octopus, orphan, abandoned, amend, stale, filtered). 8 test fixture sketches with input/expected output. `maxLane` kept implemented.
- `edgeRouter.ts` — bezier S-curve math (control points at midpoint Y), SVG path-d format, color inheritance rule (primary edges inherit parent_lane; merge fold-in edges inherit child_lane), dangling-parent stub for truncated logs, edge cases (octopus parent rows, very large lane jumps, filtered commits mid-edge). `buildShaIndex` kept implemented.
- `GraphSVG.tsx` — placeholder renderer ("Branch graph (TODO: D39-2.4b)"). Text-layout open question A (pure SVG) vs B (SVG + sibling DOM column) documented; D39-2.4b should pick B per VS Code / GitLens convention.
- `colorPalette.ts` — 10-color PALETTE finalized + `colorForIndex` (modulo wrap, defensive negatives). FILTERED_COLOR for §3.5c grey-dot rendering. Layout constants (ROW_HEIGHT=22, LANE_PITCH=16, LANE_X_OFFSET=12, COMMIT_RADIUS_VISIBLE=4, COMMIT_RADIUS_FILTERED=2) co-located so `edgeRouter` and `GraphSVG` agree.
- `interactions.ts` — `useGraphInteractions(totalRows)` hook surface (hover preview, click→diff/checkout, keyboard nav, virtualization with `@tanstack/react-virtual`).
- `integration.ts` — `useGraphData()` hook: gitSlice → memoised `{assignments, edges}` chain, memo-key contract `(commitsRef, historyFilter)`, filter-state integration per §3.5c (filter applies to `filtered_out` flag only; topology is filter-independent).

### CodeEditor wiring (ADR-036 extension)

`frontend/src/components/CodeEditor.tsx`:
- subscribes to `gitSlice.mergeInProgress.conflicted_files` — no new slice field added (the dispatch authorized adding `activeConflict` minimally, but the existing `mergeInProgress` from D39-2.3a already carries `conflicted_files`, so adding a parallel field would be drift)
- effect calls `registerConflictDecorations(editor, monaco, onAction)` when active file is in conflict state; disposes on cleanup
- the skeleton throw is wrapped in `try/catch` with `console.warn` so accidental conflict-state entry during development does not crash the editor. D39-2.4b lifts the throw and the wrapper becomes a real error report.

## Tests / build

- `cd frontend && npm run build` clean (tsc strict + vite production)
- `npx vitest run`: 31 files / 274 pass / 34 skip. No regressions vs the D39-2.3b baseline (was 27 files / 253 pass / 13 skip + 4 new files / 20 new pass / 21 new skip).
- The 21 `it.skip` cases each carry detailed test-plan docstrings + fixture sketches so D39-2.4b can flip them into passing tests without re-reading the ADR.

## Out of scope (per dispatch hard-NO list)

- backend Python (untouched)
- the 7 existing Git components from D39-2.3a/b (CommitDialog, GitHistoryList, GitDiffModal, BranchPicker, GitStatusBadge, StashApplyDialog, StashListPanel) — untouched
- `frontend/src/store/gitSlice.ts` body — we only **read** `mergeInProgress` via the existing selector; no new field added
- `frontend/src/lib/api.ts` — existing `gitMerge / gitCherryPick / gitMergeStageFile / gitMergeComplete / gitMergeAbort` wrappers already cover D39-2.4a needs
- any ADR-038 / Lineage file
- ADR docs / Toolbar.tsx

## Checklist

- [x] Checklist row updated → `docs/planning/adr-038-039-checklist.md` D39-2.4a section
- [x] CHANGELOG entry under [Unreleased] > Added
- [x] All skeleton files compile under `tsc -b` strict
- [x] All test files pass under `vitest run` (skips are deliberate)
- [x] PR body contains `Closes #941`

### CODE REVIEW

File: frontend/src/components/CodeEditor.tsx [hunk 1]
```diff
@@ -154,6 +188,48 @@ export function CodeEditor({ tab, onContentChange, onSave }: CodeEditorProps) {
...
+  // SKELETON (D39-2.4a): the effect is wired (effect runs, dispose fires
+  // on cleanup) but the `registerConflictDecorations` body throws — that
+  // throw is wrapped in a try/catch so accidental conflict-state entry
+  // during development DOES NOT crash the editor. D39-2.4b lifts the
+  // throw; the try/catch then becomes a real error path.
+  useEffect(() => {
+    if (!isInConflict) return;
+    const editor = editorRef.current;
+    const monaco = monacoRef.current;
+    if (!editor || !monaco) return;
```
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply frontend/src/components/CodeEditor.tsx:218]: accepted (fixed in 20fee29335fb087aa305606b4ab5ed08f9425b87) — added an `editorReady` state flag set inside `handleEditorMount` and added it to the effect's dep array, so the conflict-decoration effect re-runs after Monaco's lazy chunk finishes mounting. Codex P1 was correct: refs alone don't trigger re-renders, so the first-pass bail-out on null `editorRef.current` would have left decorations unregistered for the common 'open file after merge conflict' path. Verified by `npm run build` + the 7 existing CodeEditor vitest tests (all green).
